### PR TITLE
feat: --web3-notifications-widget

### PR DESCRIPTION
**Pull request by ChefBingbong**
### Summary
📲📬💱

This pull request adds the Web3Inbox feature and the Mercuryo on-ramp provider to the app. It also updates some dependencies and parameter names for consistency and readability. The Web3Inbox feature allows users to receive and manage notifications from smart contracts and dApps. The Mercuryo on-ramp provider allows users to buy crypto with fiat currency on different networks. The pull request modifies several files, such as `package.json`, `FiatOnRampModal.tsx`, `Menu/index.tsx`, `transactions/updater.tsx`, `wagmi.ts`, and `AddStableLiquidity/index.tsx`. It also adds new files, such as `NotificationWidget.tsx`, `did.json`, `web3inbox.css`, `constants.ts`, `types.ts`, `useFormatEip155Account.ts`, and `useSendNotification.ts`.

> _The app now supports Goerli and USDT_
> _And Web3Inbox for notifications, you bet_
> _With some hooks and constants_
> _And some env and DID documents_
> _The code also fixes the gas limit preset_

### Walkthrough
*  Add and configure Web3Inbox widget to send and receive push notifications ([link](https://github.com/pancakeswap/pancake-frontend/pull/7514/files?diff=unified&w=0#diff-38ac21448bcc34d3cfc05cb4c44552b636d08f9008fe911181e363c4af0be4a7R5-R7), [link](https://github.com/pancakeswap/pancake-frontend/pull/7514/files?diff=unified&w=0#diff-14b60f636e1a2b0061da57aaf231cb1ed15a5dc0c592425ed82e58fec95d42d8L54-R58), [link](https://github.com/pancakeswap/pancake-frontend/pull/7514/files?diff=unified&w=0#diff-8fb20c287d9f64663eea50168ecea60f96fd71c9b6890de5d27ec846f5aa6f8fR1-R17), [link](https://github.com/pancakeswap/pancake-frontend/pull/7514/files?diff=unified&w=0#diff-dc9fa6601376ef2b17eceb437e9ce8792468b3bec968d4751d1bbe77f9d5b8c3L1-R148), [link](https://github.com/pancakeswap/pancake-frontend/pull/7514/files?diff=unified&w=0#diff-128603fbd7fa98404a4bc03c4d7e4d2a80c1d222b84e0606b851e459a75a3793R13), [link](https://github.com/pancakeswap/pancake-frontend/pull/7514/files?diff=unified&w=0#diff-128603fbd7fa98404a4bc03c4d7e4d2a80c1d222b84e0606b851e459a75a3793R58), [link](https://github.com/pancakeswap/pancake-frontend/pull/7514/files?diff=unified&w=0#diff-0768eea2ed17afb591806d589b47841b7777221e0819cac83877030d90d7d041R1-R57), [link](https://github.com/pancakeswap/pancake-frontend/pull/7514/files?diff=unified&w=0#diff-e4bfc8d70e62e790a4ead34371ef3572c17a9fefbda6669a8b0acf7960cb9d4aR1-R39), [link](https://github.com/pancakeswap/pancake-frontend/pull/7514/files?diff=unified&w=0#diff-8b4fd2daa122b47ca1724591fc33604664233b2606ef4d1f032b39af20e1ed7fR1-R29), [link](https://github.com/pancakeswap/pancake-frontend/pull/7514/files?diff=unified&w=0#diff-8b3faec02ade75def9b135b37286560887f640c75940a6e7fac646b8d538134aR1-R9), [link](https://github.com/pancakeswap/pancake-frontend/pull/7514/files?diff=unified&w=0#diff-951bda4972529fb7ea20eb78b6de432b83980d78d004397b82c85ad75ea16a2bR1-R41), [link](https://github.com/pancakeswap/pancake-frontend/pull/7514/files?diff=unified&w=0#diff-7cf13ed80cc566159f67b4de7e56c226970328f377b6835e715068b9117aaac7L1-R24), [link](https://github.com/pancakeswap/pancake-frontend/pull/7514/files?diff=unified&w=0#diff-7cf13ed80cc566159f67b4de7e56c226970328f377b6835e715068b9117aaac7L36-R41), [link](https://github.com/pancakeswap/pancake-frontend/pull/7514/files?diff=unified&w=0#diff-7cf13ed80cc566159f67b4de7e56c226970328f377b6835e715068b9117aaac7R81), [link](https://github.com/pancakeswap/pancake-frontend/pull/7514/files?diff=unified&w=0#diff-7cf13ed80cc566159f67b4de7e56c226970328f377b6835e715068b9117aaac7L92-R97), [link](https://github.com/pancakeswap/pancake-frontend/pull/7514/files?diff=unified&w=0#diff-b746ecf7aeb22ba05a0775d1f2dfe06c2a8060b11a267f6052829f5ff4f87185R16-R22), [link](https://github.com/pancakeswap/pancake-frontend/pull/7514/files?diff=unified&w=0#diff-b746ecf7aeb22ba05a0775d1f2dfe06c2a8060b11a267f6052829f5ff4f87185L66-R75), [link](https://github.com/pancakeswap/pancake-frontend/pull/7514/files?diff=unified&w=0#diff-b746ecf7aeb22ba05a0775d1f2dfe06c2a8060b11a267f6052829f5ff4f87185L75-R83))
*  Modify Mercuryo widget parameters and URL to support Goerli test network and USDT token ([link](https://github.com/pancakeswap/pancake-frontend/pull/7514/files?diff=unified&w=0#diff-ec1dc15e7eaff4dffe6bf7d046f9b9f4c26a27a63089eee9af39f63659bb0905R28), [link](https://github.com/pancakeswap/pancake-frontend/pull/7514/files?diff=unified&w=0#diff-ec1dc15e7eaff4dffe6bf7d046f9b9f4c26a27a63089eee9af39f63659bb0905L34-R42), [link](https://github.com/pancakeswap/pancake-frontend/pull/7514/files?diff=unified&w=0#diff-ec1dc15e7eaff4dffe6bf7d046f9b9f4c26a27a63089eee9af39f63659bb0905L279-R301), [link](https://github.com/pancakeswap/pancake-frontend/pull/7514/files?diff=unified&w=0#diff-ec1dc15e7eaff4dffe6bf7d046f9b9f4c26a27a63089eee9af39f63659bb0905L346-R357), [link](https://github.com/pancakeswap/pancake-frontend/pull/7514/files?diff=unified&w=0#diff-8ea707ca06751b421a94544be240150690bd0f95b045321eaa7477bcfc33584fR46-R55), [link](https://github.com/pancakeswap/pancake-frontend/pull/7514/files?diff=unified&w=0#diff-62400b856d722556576b04bcbb9ff801cfaf142737eec302b955b8ab04a2c1f7L277-R277))
*  Change gas parameter name to gasLimit for contract function calls to match ethers dependency ([link](https://github.com/pancakeswap/pancake-frontend/pull/7514/files?diff=unified&w=0#diff-30f4e9bda6809aacb9da9383d87f8a61a75ccbdae15da50c832e913381e30fb0L121-R121), [link](https://github.com/pancakeswap/pancake-frontend/pull/7514/files?diff=unified&w=0#diff-f4ff9e7ae71a59353f06ecb29ce0023db6c69870cd4cea6fa12d4301ee52dcf1L97-R97), [link](https://github.com/pancakeswap/pancake-frontend/pull/7514/files?diff=unified&w=0#diff-d541d696a78f40e25e2fb5e5596eee09f23c12573229e2734b5bdfc1ef40b22bL311-R311), [link](https://github.com/pancakeswap/pancake-frontend/pull/7514/files?diff=unified&w=0#diff-5f12818c1d2ac2676967df71dbae86e1d21b9331dd13aca292949f5359931d8cL226-R226), [link](https://github.com/pancakeswap/pancake-frontend/pull/7514/files?diff=unified&w=0#diff-5f12818c1d2ac2676967df71dbae86e1d21b9331dd13aca292949f5359931d8cL242-R242))
*  Remove unused dependencies from `package.json` and reorder alphabetically ([link](https://github.com/pancakeswap/pancake-frontend/pull/7514/files?diff=unified&w=0#diff-14b60f636e1a2b0061da57aaf231cb1ed15a5dc0c592425ed82e58fec95d42d8L101-R113), [link](https://github.com/pancakeswap/pancake-frontend/pull/7514/files?diff=unified&w=0#diff-14b60f636e1a2b0061da57aaf231cb1ed15a5dc0c592425ed82e58fec95d42d8L150-L151), [link](https://github.com/pancakeswap/pancake-frontend/pull/7514/files?diff=unified&w=0#diff-7cf13ed80cc566159f67b4de7e56c226970328f377b6835e715068b9117aaac7L53))

### CODE REVIEW

File: apps/web/.env [hunk 1]
```diff
@@ -2,3 +2,6 @@ NEXT_PUBLIC_MM_API_URL=https://linked-pool.pancakeswap.com/quote-service
 NEXT_PUBLIC_QUOTING_API=https://swap-quoting.pancakeswap.com/quoting-service
 MOONPAY_API="https://api.moonpay.com"
 MOONPAY_PUBLISHABLE_KEY="pk_live_Ch5fat39X8NvMZwih2k7hK4sDrKanSPz"
+NEXT_PUBLIC_PROJECT_ID="ae5413feaf0cdaee02910dc807e03203"
```
Comment (chefjackson) [inline apps/web/.env:5]: Can we rename it to `NEXT_PUBLIC_NOTIFICATION_PROJECT_ID`?

File: apps/web/.env [hunk 2]
```diff
@@ -2,3 +2,6 @@ NEXT_PUBLIC_MM_API_URL=https://linked-pool.pancakeswap.com/quote-service
...
+DB_POOLING_URL="postgres://postgres:Emcg45245!!@db.wvmovalpevndzwntfwcy.supabase.co:6543/postgres"
```
Comment (chefjackson) [inline apps/web/.env:6]: We don't need it anymore?

File: pnpm-lock.yaml [hunk 3]
```diff
@@ -26287,6 +26401,38 @@ packages:
       - zod
     dev: false
 
+  /wagmi@1.3.4(encoding@0.1.13)(react-dom@18.2.0)(react@18.2.0)(typescript@5.1.3)(viem@1.2.9):
```
Comment (chefjackson) [inline pnpm-lock.yaml:26404]: Pls downgrade the pnpm version to 8.6.0 and rerun pnpm install

Comment (ChefBingbong) [reply pnpm-lock.yaml:26404]: yep willl now

File: apps/web/.env (hunk 2 above)
Comment (ChefBingbong) [reply apps/web/.env:6]: yeah we dont. cheers

File: apps/web/.env (hunk 1 above)
Comment (ChefBingbong) [reply apps/web/.env:5]: sure. renamning in next commit

**Pull request closed without merging**